Argo/Types/Decoded.swift

```diff
@@ -12,14 +12,23 @@ public enum Decoded<T> {
     default: return .None
     }
   }
+}
 
-  public static func optional<A>(p: Decoded<A>) -> Decoded<A?> {
-    switch p {
+public extension Decoded {
+  static func optional<T>(x: Decoded<T>) -> Decoded<T?> {
```

Comment:
Oh I see your point now. When I first read the return type I read it as `Decoded<T>?` so I was confused by your confusion about the name. Now I see that it's not an optional decoded T, it's a decoded optional T.